### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by chazzmoney**
Updating documentation to match expected changes for the next major release as well as make things a bit more clear.

We should make sure that we get everything we need in #49 as well.

This is NOT ready to be merged. I specifically want feedback. It also contains TODOs on the Architecture page that Danny and I need to work through together.

### CODE REVIEW

**Comment by dbanksdesign** [comment]
Sorry for the lateness on this. I'll review tomorrow

File: docs/properties.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# Definitions and Properties
+
+Properties are specific style information that you want to use in your deliverables. From a programmatic point of view, the property of size.font.small is an object containing multiple attributes, with each attribute providing specific information about size.font.small.
```
[the remark this answers is not in the document]
Comment (didoo) [reply docs/properties.md]: @dbanksdesign @chazzmoney Plus one for "design tokens". *

* I think that as maintainers/contributors we should have a "dictionary" somewhere (pun intended :) with what we should call when we write documentation and readme/examples for "design token", "property", "attribute" and "value". Personally, that's the mental schema that I tend to use:
naming

File: docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Architecture Overview
+
+This is how Style Dictionary works under the hood.
+
+## TODO
```
[the remark this answers is not in the document]
Comment (chazzmoney) [reply docs/architecture.md]: ping on this

File: docs/properties.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+Attributes are the key/object pairs in a definition. The most important (and only required) attribute is "value", which provides the data that will be used throughout the build process (and ultimately used for styling in your deliverables). You can include any custom attributes you would like (e.g. "comment" with a string or "metadata" as an object with its own attributes).
```
[the remark this answers is not in the document]
Comment (chazzmoney) [reply docs/properties.md]: we do?

File: docs/properties.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      "small" : {
+        "value": "10",
+        "comment": "the smallest font allowed for readability"
+      },
+    }
+  }
+}
+```
+
+Any property which is a string is a shorthand for that property having a value of that string.  E.g this
```
[the remark this answers is not in the document]
Comment (chazzmoney) [reply docs/properties.md]: removed

**Comment by chazzmoney** [comment]
Created #198 to replace this.

**Pull request closed without merging**